R/zzz.R

```diff
@@ -16,6 +16,35 @@ R7_object <- new_class(
 )
 methods::setOldClass("R7_object")
 
+#' @export
+`$.R7_object` <- function(x, name) {
```

Comment:
Somewhat interesting to note that if:

```r
# rather than this, where the integer vector is the core data object
new_class("foo", parent = "integer")

# we instead used this as the implementation
# (where new_class("foo", parent = "integer") would basically do this instead)
new_class("foo", properties = list(data = "integer"))
```

Then we could drop `@` altogether and use `$` to access the properties, one of which would be `$data`.

I know you mentioned that we probably want a way to directly build on base atomic objects, so it probably is too late, but it is interesting to think about how `@` could be removed (that would probably make autocomplete work on properties out of the box too, probably through `.DollarNames`)

Reply:
I guess you still have to think about whether or not you want this with lists though

```
# if this:
new_class("foo", parent = "list")

# was basically this:
foo <- new_class("foo", properties = list(data = "list")

obj <- foo(list(x = 1))

# then you'd have to do this:
obj$data$x
```

Reply:
I think the problem is then that `is.list()` would be TRUE for R7 objects which seems likely to lead to a bunch of unintended downstream consequences.

Reply:
I was still thinking the core object would be that weird S4 gremlin thing that is returned by `R7_object()`, and we'd still have all the properties as attributes on that